x-pack/plugins/maps/server/maps_telemetry/maps_telemetry.ts

```diff
@@ -299,30 +303,58 @@ export function buildMapsTelemetry({
     },
   };
 }
-async function getMapSavedObjects(
-  savedObjectsClient: SavedObjectsClientContract | ISavedObjectsRepository
-) {
-  const mapsSavedObjects = await savedObjectsClient.find<MapSavedObjectAttributes>({
-    type: MAP_SAVED_OBJECT_TYPE,
-  });
-  return mapsSavedObjects.saved_objects;
-}
 
-async function getIndexPatternSavedObjects(
-  savedObjectsClient: SavedObjectsClientContract | ISavedObjectsRepository
+export async function execTransformOverMultipleSavedObjectPages<T>(
+  savedObjectType: string,
+  transform: (savedObjects: Array<SavedObject<T>>) => void
 ) {
-  const indexPatternSavedObjects = await savedObjectsClient.find<IndexPatternAttributes>({
-    type: 'index-pattern',
-  });
-  return indexPatternSavedObjects.saved_objects;
+  const savedObjectsClient = getInternalRepository();
+
+  let currentPage = 1;
+  // Seed values
+  let page = 0;
+  let perPage = 0;
+  let total = 0;
+  let savedObjects = [];
+
+  do {
+    const savedObjectsFindResult = await savedObjectsClient.find<T>({
+      type: savedObjectType,
+      page: currentPage++,
+    });
+    ({ page, per_page: perPage, saved_objects: savedObjects, total } = savedObjectsFindResult);
+    transform(savedObjects);
+  } while (page * perPage < total);
 }
 
-export async function getMapsTelemetry(config: MapsConfigType) {
-  const savedObjectsClient = getInternalRepository();
-  const mapSavedObjects = await getMapSavedObjects(savedObjectsClient);
-  const indexPatternSavedObjects = await getIndexPatternSavedObjects(savedObjectsClient);
-  const settings = {
-    showMapVisualizationTypes: config.showMapVisualizationTypes,
+export async function getMapsTelemetry(config: MapsConfigType): Promise<MapsUsage> {
+  // Get layer descriptors for Maps saved objects. This is not set up
+  // to be done incrementally (i.e. - per page) but minimally we at least
+  // build a list of small footprint objects
+  let layerLists: LayerDescriptor[][] = [];
+  await execTransformOverMultipleSavedObjectPages<MapSavedObjectAttributes>(
+    MAP_SAVED_OBJECT_TYPE,
+    (savedObjects) => (layerLists = layerLists.concat(getLayerLists(savedObjects)))
```

Comment:
Nit: How about just using push instead of concat so layerLists can be a const.

`(savedObjects) => (layerLists.push(getLayerLists(savedObjects)))`

Also another nit, `getLayerLists` should be named `getLayerList` because it return a single layerList from a map. It does not return plural "lists".

Reply:
Nevermind, I see now that getLayerLists is processing an array of saved objects. double arrays always hurt my head.

Reply:
> Nit: How about just using push instead of concat so layerLists can be a const.

Agreed. Will update!

> Nevermind, I see now that getLayerLists is processing an array of saved objects. arrays of arrays always hurt my head.

Yeah, mine too